app/actors/DBPollActor.scala

```diff
@@ -58,7 +58,7 @@ abstract class DBPollActor(schema: String = "public", table: String) extends Pol
   private def safeProcessRecord(record: ProcessQueueDelivery): Unit = {
     Try {
       logger.info("Inside safeProcessRecord method")
-      process(record)
+      process[ProcessQueueDelivery](record)
```

Comment:
This method will be called for every actor which will have a different type. You cannot hardcode the type here. This is still not generic. As me and Prakhar said - making it generic won't be straight forward.

Reply:
Okay, I'm looking into that.